### PR TITLE
shareTarget derives full action URL

### PR DESCRIPTION
**Pull request by andreban**
- Derive the full action URL from the Web Manifest URL.
- Extracts mime-types from share-target.

Improves #21

### CODE REVIEW

File: packages/core/src/lib/TwaManifest.ts [hunk 1]
```diff
@@ -294,15 +294,30 @@ export class TwaManifest {
...
-      // https://github.com/GoogleChromeLabs/bubblewrap/issues/21.
-      shareTarget: undefined,
+      shareTarget: TwaManifest.verifyShareTarget(webManifestUrl, webManifest.share_target),
     });
     return twaManifest;
   }
 
+  private static verifyShareTarget(
+      webManifestUrl: URL, shareTarget?: ShareTarget): ShareTarget | undefined {
+    if (!shareTarget || !shareTarget.action || !shareTarget.params || !shareTarget.params.files) {
```
Comment (PEConn) [inline packages/core/src/lib/TwaManifest.ts:304]: Can this not be `if (!shareTarget?.action || !shareTarget.params?.files)`?

**Review by PEConn — COMMENTED** [review]
I think this looks good - I'm a little unsure about why we're using undefined instead of null though.

File: packages/core/src/lib/TwaManifest.ts (hunk 1 above)
Comment (andreban) [reply packages/core/src/lib/TwaManifest.ts]: Good point. done.

**Comment by andreban** [comment]
Unfortunately, when a field is defined as optional like `shareTarget? : ShareTarget`), that means `ShareTarget | undefined` and returning `null` causes a compilation error. (`Type 'ShareTarget | null' is not assignable to type 'ShareTarget | undefined'.`)

File: packages/core/src/lib/TwaManifest.ts [hunk 2]
```diff
@@ -294,15 +294,30 @@ export class TwaManifest {
...
-      // Target implementation is being tracked at
-      // https://github.com/GoogleChromeLabs/bubblewrap/issues/21.
-      shareTarget: undefined,
+      shareTarget: TwaManifest.verifyShareTarget(webManifestUrl, webManifest.share_target),
     });
     return twaManifest;
   }
 
+  private static verifyShareTarget(
+      webManifestUrl: URL, shareTarget?: ShareTarget): ShareTarget | undefined {
```
Comment (PEConn) [inline packages/core/src/lib/TwaManifest.ts:303]: The return type can be `ShareTarget?` right?

**Review by PEConn — APPROVED** [review]
Looks good with nit.

File: packages/core/src/lib/TwaManifest.ts (hunk 2 above)
Comment (andreban) [reply packages/core/src/lib/TwaManifest.ts:303]: The optional syntax only works for attributes, variable declarations and parameters. Not for return types.

**Comment by PEConn** [comment]
Oh, it seems I still don't understand TypeScript. LGTM again.

**Pull request merged**